auth/opa/rpcauth/redact.go

```diff
@@ -120,7 +123,21 @@ func getRedactedInput(input *RPCAuthInput) (RPCAuthInput, error) {
 		if err := protojson.Unmarshal([]byte(input.Message), redactedMessage); err != nil {
 			return RPCAuthInput{}, fmt.Errorf("could not marshal input into %v: %v", input.MessageType, err)
 		}
-		redactFields(redactedMessage.ProtoReflect())
+		if proxyReq, ok := redactedMessage.(*proxypb.ProxyRequest); ok && proxyReq.GetStreamData() != nil {
```

Comment:
Could you add a test for this?

(I'm mildly surprised that casting redactedMessage to a *proxypb.ProxyRequest works, I didn't realize that the concrete type was used when doing this reflection)

Reply:
Altenatively, maybe it's better to do the anypb checks inside redactFields() and make them work for any proto containing anypbs?

Reply:
Great suggestion.
Added a test and moved the anypb check inside redactNestedField() -- now it works with any proto containing anypb :D!